### PR TITLE
Time to reserve

### PR DESCRIPTION
**Pull request by xinuc**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
 source 'https://rubygems.org'
 
+gem 'tcp_timeout', git: "https://github.com/bukalapak/tcp_timeout.git"
```
Comment (contentfree) [inline Gemfile]: I don't think we want to have these dependencies added here.

**Comment by contentfree** [comment]
I see you closed this PR. There's stuff in here that would be beneficial to Backburner. Maybe split the PR up into several, more-specific PRs.

**Comment by xinuc** [comment]
Hi @contentfree,
sorry, this PR should be for our own fork, not upstream.

Github makes it too easy to create PR for upstream.

**Comment by contentfree** [comment]
We're certainly open to some upstream PRs. Especially the current thread storage of the connection. We'd like to hear your reasoning on keeping connections open vs keeping them cleaned up.

**Comment by contentfree** [comment]
True true, @eltone

**Pull request closed without merging**